iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnBuffers.cpp

```diff
@@ -853,13 +836,30 @@ struct MatmulOnTensorConversion
                       ArrayRef<Value> resultBuffers,
                       ConversionPatternRewriter &rewriter) const {
     if (!op.hasTensorSemantics()) return failure();
-    // The last one is a init tensor.
     rewriter.create<LinalgOpTy>(
         op.getLoc(), inputBuffers.drop_back(op.getNumResults()), resultBuffers);
     return success();
   }
 };
 
+/// Converts linalg.conv-ish on tensors to linalg.conv-ish on buffers.
```

Comment:
This is fine for now. But the `LinalgBufferize.cpp` in `Conversion/Common/` has a generic lowering for any Linalg structured op. Maybe we should just used that and remove a whole bunch of code here? The advantage is that it propagates attributes too.

Reply:
This will need some refactoring since the methods are all static. Since the change is big, I would like to do it in a follow up. I copied a bit code from there, and only leave one conversion pattern.